wordpress-mobile/WordPress-iOS
# Fetch plugin icons from the WordPress.org API

This introduces the Remote code to get plugin information from api.wordpress.org and get the icon for each plugin.

Some unusual things in this PR:
- I've used the new Swift 4 `Decodable` and `JSONDecoder` to parse the response. It's pretty straightforward and probably the best way of doing JSON from now on, but it's different from what we do everywhere else.
- Since this is a new API and can't reuse the existing clients. I've used `URLSession` directly. We still have to figure out if we want to start using `Alamofire` or what happens to our network architecture, but since it's just one call for now, I went for the simple barebones solution.

See #7471

To test:

- Load the Plugins section for a site
- Verify that icons are loaded for each plugin that should have one
